BuildToolsVersion.txt

```diff
@@ -1 +1 @@
-2.0.0-prerelease-01708-02
\ No newline at end of file
+2.0.0-prerelease-01714-04
```

Comment:
https://github.com/dotnet/buildtools/pull/1572

CoreClr is the first repo to consume BuildTools after a recent refactoring and is the only repo making use of BlobNamePrefix to trim down to a specific filename prefix (other repos are either not using BlobNamePrefix or use it to trim to a specific folder within a container).  The fix made to buildtools is to ensure that we trim folder structure on a folder level, and still maintain compatibility with how other repo's are using the task.  

I've validated this locally for common coreclr, corefx, and core-setup usage of the downloadfromazure task.